url0d084b4168_zhourui392_BaiduPicDown_tgz-pJ8-org_json_TestJ8/src/zz/spider/Spider.java

```diff
@@ -0,0 +1,73 @@
+package zz.spider;
+
+import java.io.File;
+
+import org.nutz.http.Http;
+import org.nutz.http.Response;
+import org.nutz.lang.Strings;
+
+import zz.config.ZZConfig;
+import zz.imageweb.BaiduImage;
+import zz.imageweb.ImageInterface;
+
+public class Spider  implements Runnable {
+	private String preHtml;
+     
+    private String mSavePath;
+     
+    public Spider(String savePath) {
+        setmSavePath(savePath);
+    }
+     
+	@Override
+    public void run() {
+        String html ;
+         
+        ImageInterface url = new BaiduImage();		//ImageInterface接口，自由替换
+        for(;;) {
+        	url.init();
+            try {
+            	if (Strings.isEmpty(url.getHtmlURL())){
+            		break;
+            	}
+            	Response resp = Http.get(url.getHtmlURL());
+            	if (resp.isOK()) {
+            	    try {
+                        html = resp.getContent();
+                        if(html.equals(preHtml)) {
+                            continue;
+                        }
+                         
+                        preHtml = html;
+                        
+                        File savePathDir = new File(new File(ZZConfig.loaclFileDir).getAbsolutePath() +"\\" + url.getDirName());
+                        
+                        if (!savePathDir.exists()){
+                        	savePathDir.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)